bd-workflows/src/engine.rs

```diff
@@ -586,9 +628,17 @@ impl WorkflowsEngine {
         self.needs_state_persistence = true;
       }
 
-      let (triggered_actions, workflow_logs_to_inject) = result.into_parts();
-      actions.extend(triggered_actions);
+      // In debug only mode we do not trigger any actions, but we still inject logs so that
+      // workflows continue to advance if they depend on the injected logs.
```

Comment:
Agreed that this is required for correctness, though interesting how we do end up having debug workflows have a real impact on the timeline data